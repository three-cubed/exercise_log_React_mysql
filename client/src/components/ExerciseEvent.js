import React from 'react';

const ExerciseEvent = (props) => {
    const deleteExercise = async (event, exerciseEventId, callback) => {
        event.preventDefault();
        await fetch(`/delete/${exerciseEventId}`, {
            method: 'delete',
        });
        callback();
    };
    return (
        <>
            <div className='exerciseEvent' data-testid='exerciseEvent' style={{ color: props.colour, borderLeft: 'solid 16px', borderBottom: 'solid 2px' }}>
                <p> 
                    &emsp; <small>{props.exerciseDate}</small><br />
                    &emsp; <b>{props.exerciseEventTitle}</b>&emsp;&emsp;&emsp; <br />
                    <button 
                        className='deletionBtn' 
                        data-testid='deletionBtn' 
                        onClick={event => deleteExercise(event, props.id, props.restateAfterDelete)}
                    > 
                    X
                    </button>
                    <span className='achievementSpan'>
                        &emsp; {props.achievement}
                    </span>
                </p>
            </div>
            <br />
        </>
    )
}

export default ExerciseEvent;
