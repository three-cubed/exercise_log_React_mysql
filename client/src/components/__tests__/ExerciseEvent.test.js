import React from 'react';
import { fireEvent, render, cleanup } from '@testing-library/react';
import { toHaveStyle } from '@testing-library/jest-dom'; 
import renderer from 'react-test-renderer';
import ExerciseEvent from '../ExerciseEvent.js';

afterEach(cleanup);

describe('ExerciseEvent component', () => {
    test('renders', () => {
        render(<ExerciseEvent />);
    });
    test('matches snapshot', () => {
        const tree = renderer.create(<ExerciseEvent 
                                        exerciseDate={'99th Octember'} 
                                        exerciseEventTitle={'dummy event title'} 
                                        colour={'blue'} />)
                                    .toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('shows exerciseDate which was entered as prop', () => {
        const newExerciseEvent = render(<ExerciseEvent exerciseDate={'dummy date'} />);
        expect(newExerciseEvent.getByText(/dummy date/));
    });
    test('shows exerciseEventTitle which was entered as prop', () => {
        const newExerciseEvent = render(<ExerciseEvent exerciseEventTitle={'dummy event title'} />);
        expect(newExerciseEvent.getByText(/dummy event title/));
    });
    test('shows achievement which was entered as prop', () => {
        const newExerciseEvent = render(<ExerciseEvent achievement= {'dummy achievement'}/>);
        expect(newExerciseEvent.getByText(/dummy achievement/));
    });
    test('shows colour which was entered as prop', () => {
        const newExerciseEvent = render(<ExerciseEvent colour={'blue'} />);
        expect(newExerciseEvent.getByTestId('exerciseEvent')).toHaveStyle('color: blue');
    });
    test('pressing deletionBtn sets off function', () => {
        const performFunction = jest.fn();
        const newExerciseEvent = render(<ExerciseEvent />);
        const deletionBtn = newExerciseEvent.getByTestId('deletionBtn');
        deletionBtn.onclick = performFunction;
        fireEvent.click(deletionBtn);
        expect(performFunction).toBeCalled();
    });
})
