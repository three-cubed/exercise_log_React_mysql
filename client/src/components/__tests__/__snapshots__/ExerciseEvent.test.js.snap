// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseEvent component matches snapshot 1`] = `
Array [
  <div
    className="exerciseEvent"
    data-testid="exerciseEvent"
    style={
      Object {
        "borderBottom": "solid 2px",
        "borderLeft": "solid 16px",
        "color": "blue",
      }
    }
  >
    <p>
        
      <small>
        99th Octember
      </small>
      <br />
        
      <b>
        dummy event title
      </b>
          
      <br />
      <button
        className="deletionBtn"
        data-testid="deletionBtn"
        onClick={[Function]}
      >
        X
      </button>
      <span
        className="achievementSpan"
      >
          
      </span>
    </p>
  </div>,
  <br />,
]
`;
